VideoApi/VideoApi/Controllers/MagicLinksController.cs

```diff
@@ -0,0 +1,110 @@
+        private readonly ILogger<MagicLinksController> _logger;
+
+        public MagicLinksController(ICommandHandler commandHandler, IQueryHandler queryHandler, IMagicLinksJwtTokenProvider magicLinksJwtTokenProvider, ILogger<MagicLinksController> logger)
+        {
+            _commandHandler = commandHandler;
+            _queryHandler = queryHandler;
+            _magicLinksJwtTokenProvider = magicLinksJwtTokenProvider;
+
+            _logger = logger;
+        }
+
+        [HttpGet("ValidateMagicLink/{hearingId}")]
+        [OpenApiOperation("ValidateMagicLink")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> ValidateMagicLink(Guid hearingId)
+        {
+            try
+            {
+                var query = new GetConferenceByHearingRefIdQuery(hearingId);
+                var conference =
+                    await _queryHandler.Handle<GetConferenceByHearingRefIdQuery, Conference>(query);
+
+                var isMagicLinkValid = MagicLink.IsValid(conference);
+                
+                return Ok(isMagicLinkValid);
+            }
+            catch (ConferenceNotFoundException ex)
+            {
+                _logger.LogError(ex, "Unable to find conference");
+                return NotFound(false);
+            }
+        }
+
+        [HttpPost("AddMagicLinkParticipant/{hearingId}")]
+        [OpenApiOperation("AddMagicLinkParticipant")]
+        [ProducesResponseType(typeof(AddMagicLinkParticipantResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> AddMagicLinkParticipant(Guid hearingId, AddMagicLinkParticipantRequest magicLinkParticipantRequest)
+        {
+            try
+            {
+                var query = new GetConferenceByHearingRefIdQuery(hearingId);
+                var conference =
+                    await _queryHandler.Handle<GetConferenceByHearingRefIdQuery, Conference>(query);
+
+                var participant = new MagicLinkParticipant(magicLinkParticipantRequest.Name, magicLinkParticipantRequest.UserRole.MapToDomainEnum());
+
+                var participantsToAdd = new List<ParticipantBase> { participant };
+
+                var addParticipantsToConferenceCommand = new AddParticipantsToConferenceCommand(conference.Id, participantsToAdd, new List<LinkedParticipantDto>());
+
+                await _commandHandler.Handle(addParticipantsToConferenceCommand);
+
+                var jwtDetails = _magicLinksJwtTokenProvider.GenerateToken(participant.Name, participant.Username,
+                    participant.UserRole);
+
+                var addMagicLinkParticipantTokenCommand = new AddMagicLinkParticipantTokenCommand(participant.Id, jwtDetails);
+                
+                await _commandHandler.Handle(addMagicLinkParticipantTokenCommand);
```

Comment:
I am very new with cqrs pattern so I might be wrong, but does this endpoint need to refer to 3 command/queries here? Is that the approach suggested to apply this pattern or should we include all these inside of a command?